src/ucp/core/ucp_worker.c

```diff
@@ -249,7 +249,13 @@ static void ucp_ep_err_pending_purge(uct_pending_req_t *self, void *arg)
     ucp_request_t *req = ucs_container_of(self, ucp_request_t, send.uct);
     ucs_status_t status = (uintptr_t)arg;
 
-    ucp_request_complete_send(req, status);
+    if (req->send.uct_comp.func) {
+        if (--req->send.uct_comp.count == 0) {
```

Comment:
This looks like something that should already be done by UCT. Isn't this the case?